DistributedLock.Tests/Infrastructure/Redis/TestingRedisDatabaseProvider.cs

```diff
@@ -22,17 +24,35 @@ protected TestingRedisDatabaseProvider(int count)
         {
         }
 
+        protected TestingRedisDatabaseProvider(int count, string keyPrefix)
+            : this(Enumerable.Range(0, count).Select(i => RedisServer.GetDefaultServer(i).Multiplexer.GetDatabase().WithKeyPrefix(keyPrefix)))
+        {
+        }
+
         // publicly settable so that callers can alter the dbs in use
         public IReadOnlyList<IDatabase> Databases { get; set; }
 
         public virtual string CrossProcessLockTypeSuffix => this.Databases.Count.ToString();
     }
 
+    public interface ITestingRedisWithKeyPrefixDatabaseProvider

```

Comment:
Why is this needed?

Reply:
The need to run RunLockTaker / DistributedLockTaker with global lock name (prefix + local name) in order to pass cross process tests.